src/app/pages/signature-request/components/message-box.tsx

```diff
@@ -0,0 +1,51 @@
+import { color, Stack, Text } from '@stacks/ui';
+import { sha256 } from 'sha.js';
+import { HashDrawer } from './hash-drawer';
+import { useEffect, useState } from 'react';
+
+interface MessageBoxProps {
+  message: string;
+}
+export function MessageBox(props: MessageBoxProps): JSX.Element | null {
+  const { message } = props;
+  const [hash, setHash] = useState<string | undefined>();
+  useEffect(() => {
+    setHash(new sha256().update(message).digest('hex'));
+  }, [message]);
```

Comment:
Could the result of this be passed in to the component? This seems like core logic to the feature, so reckon it'd better belong higher up the tree.

Reply:
This is just for display that's why I placed it in the component. The hash used for signature is calculated internally in stacks.js.